common/meta/grpc/src/grpc_client.rs

```diff
@@ -390,7 +429,10 @@ impl MetaGrpcClient {
                     let ch = self.conn_pool.get(addr).await;
                     match ch {
                         Ok(c) => {
-                            *current_endpoint = Some(addr.clone());
+                            {
+                                let mut current_endpoint = self.current_endpoint.write().await;
+                                *current_endpoint = Some(addr.clone());
```

Comment:
If current_endpoint  does not need to be locked across await, using a std::sync::Mutex would be more efficient.